### DISCUSSION TITLE
Fabric with pre-shipped java or oracle?

### DISCUSSION THREAD

**Discussion by SoundEngineDevelopement** [FabricMC/fabric-api#1892] [Mod Dev Support]
Am i supposed to use pre-shipped java with minecraft fabric or use the one i installed from oracle?
Rumors say oracle java performs better. With Optifine i can use oracle java but when i try it with fabric it says i must use the pre-installed java version.
Why is that? Is it possible to use oracle java with fabric? If performance is better i would use oracle java if possible.

**Comment by frqnny** [comment]
this must be the JavaCheck.java jar bug. 

try this video? https://youtu.be/-0U1jSaEueo

**Reply by SoundEngineDevelopement** [reply]
> this must be the JavaCheck.java jar bug.
> 
> try this video? https://youtu.be/-0U1jSaEueo

So i did what he said in the video, under C:\Program Files (x86)\Minecraft Launcher\game i renamed this file "JavaCheck.jar" to "JavaCheck_disabled.jar" but the result was an instant crash, then i deleted the file with the same result, instant crash,
Here is the error: https://i.postimg.cc/x16nkC76/error.png

**Comment by SoundEngineDevelopement** [comment]
Maybe i just have to wait for a newer version of fabric? or minecraft? or java? Hell i dont know but my best bet is to wait for new fabric.
So fixing the "Java Runtime Version Incompatible error" only bourght me another error instead.
Seems as this fix is currently not working.

**Comment by warjort** [comment]
The screen shot you linked to says you are trying to run with java 8?
Minecraft 1.18 needs a version of java 17

> Maybe i just have to wait for a newer version of fabric? or minecraft? or java? Hell i dont know

The underlying issue referred to above  is with none of these, it is with the minecraft launcher.
https://bugs.mojang.com/browse/MCL-18306?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true

**Reply by SoundEngineDevelopement** [reply]
> The screen shot you linked to says you are trying to run with java 8? Minecraft 1.18 needs a version of java 17
> 
> > Maybe i just have to wait for a newer version of fabric? or minecraft? or java? Hell i dont know
> 
> The underlying issue referred to above is with none of these, it is with the minecraft launcher. https://bugs.mojang.com/browse/MCL-18306?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true

I use optifine 1.18 with java 8 and it works. However with fabric it does not work and i started using java 17 which is working.
But.. java 8 had a config option via the windows control panel, with java 17 i cannot find such a control panel.
Why is that? Where is the control panel for java 17?

**Comment by warjort** [comment]
> Why is that? Where is the control panel for java 17?

It was removed in java 11. 

https://docs.oracle.com/en/java/javase/16/migrate/removed-tools-and-components.html#GUID-460800EF-A523-4B10-B694-E3536AC419C1

That panel was for controlling how java runs in your browser which isn't very relevant for modern minecraft or web browsers.

**Reply by SoundEngineDevelopement** [reply]
> > Why is that? Where is the control panel for java 17?
> 
> It was removed in java 11.
> 
> https://docs.oracle.com/en/java/javase/16/migrate/removed-tools-and-components.html#GUID-460800EF-A523-4B10-B694-E3536AC419C1
> 
> That panel was for controlling how java runs in your browser which isn't very relevant for modern minecraft or web browsers.

So does that mean that Java 17 runs in my browser without the possibility to disable it or does it not run at all?
I like to have configuration options, unless there is nothing to configure??

**Reply by Juuxel** [reply]
Java hasn't had any sort of web browser support for years since applets and java web start were removed in Java 11, so Java 17 doesn't run in your browser at all.

**Comment by warjort** [comment]
> So does that mean that Java 17 runs in my browser without the possibility to disable it or does it not run at all?

Unless you are using an old browser (which would probably need an old pre java 11 runtime anyway) 
or you are using a separate browser plugin (which would have its own configuration to enable it)
java 17 won't be enabled in your browser.

Java 17 could still technically run inside a browser, but the api(s) to handle this are going away in future:
(applets) https://bugs.openjdk.java.net/browse/JDK-8256145
(security manager) https://bugs.openjdk.java.net/browse/JDK-8264713
